feat: add `BitVec.clzAuxRec_eq_clzAuxRec_of_getLsbD_false`

luisacicolini opened the pull request:
This PR adds theorem `BitVec.clzAuxRec_eq_clzAuxRec_of_getLsbD_false` as a more general statement than `BitVec.clzAuxRec_eq_clzAuxRec_of_le`, replacing the latter in the bitblaster too.

luisacicolini commented on src/Std/Tactic/BVDecide/Bitblast/BVExpr/Circuit/Lemmas/Operations/Clz.lean:
```diff
@@ -57,7 +57,7 @@ theorem go_denote_eq {w : Nat} (aig : AIG α)
           · simp [Ref.hgate]
         · intro idx hidx
-          simp only [Nat.add_eq_zero, Nat.succ_ne_self, and_false, reduceIte, 
+          simp only [Nat.add_eq_zero, Nat.succ_ne_self, and_false, reduceIte,
```
```suggestion
          simp only [Nat.add_eq_zero, Nat.succ_ne_self, and_false, reduceIte, 
```

bollu commented on src/Init/Data/BitVec/Lemmas.lean:
```diff
@@ -5761,15 +5761,21 @@ theorem clzAuxRec_zero (x : BitVec w) :
-    x.clzAuxRec n = x.clzAuxRec (w - 1) := by
-  let k := n - (w - 1)
-  rw [show n = (w - 1) + k by omega]
+theorem clzAuxRec_eq_clzAuxRec_of_getLsbD_false (x : BitVec w) (h : ∀ i, n < i → x.getLsbD i = false) :
```
```suggestion
theorem clzAuxRec_eq_clzAuxRec_of_getLsbD_false {x : BitVec w} (h : ∀ i, n < i → x.getLsbD i = false) :
```

luisacicolini commented:
thanks @bollu, I addressed your suggestions and also re-added the old lemma (and used that one in the bitblaster, since it simplifies the proof by quite a bit), so this PR only contains the general version of the lemma.

luisacicolini commented:
changelog-library
